lib/pure/osproc.nim

```diff
@@ -1544,16 +1544,52 @@ elif not defined(useNimRtl):
 
     result = int(select(cint(m+1), addr(rd), nil, nil, nil)) == 1
 
+proc execCmdExImpl(command: string, options: set[ProcessOption], env: StringTableRef,
+    workingDir: string, input: string, args: openArray[string], useArgs: bool): tuple[
+    output: TaintedString,
+    exitCode: int] {.tags: [ExecIOEffect, ReadIOEffect, RootEffect], gcsafe.} =
+  when (NimMajor, NimMinor, NimPatch) < (1, 3, 5):
+    doAssert input.len == 0
+    doAssert workingDir.len == 0
+    doAssert env == nil
+  var options = options
+  if args.len == 0 and not useArgs:
+    options.incl poEvalCommand
+  var p = startProcess(command, options = options,
+    workingDir = workingDir, env = env, args = args)
+  var outp = outputStream(p)
+
+  if input.len > 0:
+    # There is no way to provide input for the child process
+    # anymore. Closing it will create EOF on stdin instead of eternal
+    # blocking.
+    # Writing in chunks would require a selectors (eg kqueue/epoll) to avoid
+    # blocking on io.
+    inputStream(p).write(input)
+  close inputStream(p)
+
+  result = (TaintedString"", -1)
+  var line = newStringOfCap(120).TaintedString
+  while true:
+    if outp.readLine(line):
+      result[0].string.add(line.string)
+      result[0].string.add("\n")
+    else:
+      result[1] = peekExitCode(p)
+      if result[1] != -1: break
+  close(p)
 
 proc execCmdEx*(command: string, options: set[ProcessOption] = {
                 poStdErrToStdOut, poUsePath}, env: StringTableRef = nil,
-                workingDir = "", input = ""): tuple[
+                workingDir = "", input = "", args: openArray[string]): tuple[
                 output: TaintedString,
                 exitCode: int] {.tags:
-                [ExecIOEffect, ReadIOEffect, RootEffect], gcsafe.} =
+                [ExecIOEffect, ReadIOEffect, RootEffect], gcsafe, inline, since: (1,5,1).} =
   ## A convenience proc that runs the `command`, and returns its `output` and
-  ## `exitCode`. `env` and `workingDir` params behave as for `startProcess`.
+  ## `exitCode`. `env`, `workingDir`, `args` params behave as for `startProcess`.
   ## If `input.len > 0`, it is passed as stdin.
+  ## When `args` is not specified, another overload is used that treats `command`
```

Comment:
What? That's terrible... ``poEvalCommand`` is already a security nightmare and you use it implicitly, depending on how this thing is called? Not to mention that "another overload is used" is an irrelevant implementation detail that might as well be misunderstood as "the Nim compiler picks a different overload" which is impossible.

Reply:
this has been resolved via `execArgs`